### PR TITLE
add rentable_by filter 

### PR DESCRIPTION
**Pull request by Omarabdul3ziz**
- implement rentable_by filter for node to return nodes already rented by a twin id or nodes available to rent.
- https://github.com/threefoldtech/tfgrid-sdk-go/issues/1078

### Checklist

- [ ] Tests included
- [ ] Build pass
- [ ] Documentation
- [ ] Code format and docstring

### CODE REVIEW

File: grid-proxy/pkg/types/nodes.go [hunk 1]
```diff
@@ -141,7 +141,8 @@ type NodeFilter struct {
 	OwnedBy           *uint64  `schema:"owned_by,omitempty"`
 	Rented            *bool    `schema:"rented,omitempty"`
 	RentedBy          *uint64  `schema:"rented_by,omitempty"`
-	AvailableFor      *uint64  `schema:"available_for,omitempty"`
+	RentableBy        *uint64  `schema:"rentable_by,omitempty"`   // rented by twin or rentable
```
Comment (xmonader) [inline grid-proxy/pkg/types/nodes.go]: RentedBy instead of RentableBy no?

Comment (Omarabdul3ziz) [reply grid-proxy/pkg/types/nodes.go]: `rentedby` is already a functional filter returns all the nodes rented by a specific twin.
`rentableby` is a new filter that should return all the nodes rented by a specific twin + all the nodes available to rent

Comment (Omarabdul3ziz) [reply grid-proxy/pkg/types/nodes.go]: `rentedby` is already a functional filter returns all the nodes rented by a specific twin.
`rentableby` is a new filter that should return all the nodes rented by a specific twin + all the nodes available to rent. somehow combining `rentable` and `rentedby`

Comment (xmonader) [reply grid-proxy/pkg/types/nodes.go]: How is that different from AvailableFor?

Comment (Omarabdul3ziz) [reply grid-proxy/pkg/types/nodes.go]: `available_for` returns the nodes rented by a twin id + the shared nodes. (available to deploy on)
`rentable_for` returns the nodes rented by a twin id + the rentable nodes. (rented or available to rent)

Comment (xmonader) [reply grid-proxy/pkg/types/nodes.go]: Tb bos :D let's make it rentable_or_rented_by :D

**Review by ashraffouda — APPROVED** [review]

**Pull request merged**